product_tag_pricelist/models/product_pricelist_item.py

```diff
@@ -0,0 +1,110 @@
+
+    @api.depends(
+        "applied_on",
+        "categ_id",
+        "product_tmpl_id",
+        "product_id",
+        "compute_price",
+        "fixed_price",
+        "pricelist_id",
+        "percent_price",
+        "price_discount",
+        "price_surcharge",
+        "product_tag_ids",
+    )
+    def _compute_name_and_price(self):
+        res = super()._compute_name_and_price()
+        for item in self:
+            if item.product_tag_ids and item.applied_on == "2a_product_tags":
+                tag_names = item.product_tag_ids.mapped("name")
+                tag_names_str = ", ".join(tag_names)
+                item.name = _("Tags: {}").format(tag_names_str)
+        return res
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for values in vals_list:
+            if not values.get("applied_on") and values.get("product_tag_ids"):
+                values["applied_on"] = "2a_product_tags"
+            if values.get("applied_on"):
+                # Ensure item consistency for later searches.
+                applied_on = values["applied_on"]
+                if applied_on == "3_global":
+                    values.update(dict(product_tag_ids=False))
+                elif applied_on == "2a_product_tags":
+                    values.update(
+                        dict(product_id=None, product_tmpl_id=None, categ_id=None)
+                    )
+                elif applied_on == "2_product_category":
+                    values.update(dict(product_tag_ids=False))
+                elif applied_on == "1_product":
+                    values.update(dict(product_tag_ids=False))
+                elif applied_on == "0_product_variant":
+                    values.update(dict(product_tag_ids=False))
+        return super().create(vals_list)
+
+    def write(self, values):
+        if values.get("applied_on"):
+            applied_on = values["applied_on"]
+            if applied_on == "3_global":
+                values.update(dict(product_tag_ids=False))
+            elif applied_on == "2a_product_tags":
+                values.update(
+                    dict(product_id=None, product_tmpl_id=None, categ_id=None)
+                )
+            elif applied_on == "2_product_category":
+                values.update(dict(product_tag_ids=False))
+            elif applied_on == "1_product":
+                values.update(dict(product_tag_ids=False))
+            elif applied_on == "0_product_variant":
+                values.update(dict(product_tag_ids=False))
+        return super().write(values)
+
+    def _is_applicable_for(self, product, qty_in_product_uom):
+        self.ensure_one()
+        product.ensure_one()
+
+        if (
+            not (self.min_quantity and qty_in_product_uom < self.min_quantity)
+            and self.applied_on == "2a_product_tags"
+        ):
+            p_tags = product.product_tag_ids
+            if product._name == "product.product":
```

Comment:
Please, also consider product.template.

Reply:
It is considered, with product templates it will look for the product_tag_ids field.

Product template does not have the additional_product_tag_ids field

Reply:
Yes, you're right.